mPowerSDK/APHTasksReminderManager.m

```diff
@@ -0,0 +1,156 @@
+        // a notification today if one or more have.
+        
+        NSDate *today = [NSDate date];
+        NSDate *yesterday = today.dayBefore;
+        
+        // creating and using a filter here eventhough we don't need one because we're looking for all
+        // the tasks. But, the fetch method in APCScheduler that does not require a filter is crashing
+        // because filter is nil, so it seem there's a bug there
+        
+        NSPredicate *filterForallTasks = [NSPredicate predicateWithFormat: @"%K == nil || %K == %@ || %K == %@",
+                                          NSStringFromSelector(@selector(taskIsOptional)),
+                                          NSStringFromSelector(@selector(taskIsOptional)),
+                                          @(NO),
+                                          NSStringFromSelector(@selector(taskIsOptional)),
+                                          @(YES)];
+
+        [[APCScheduler defaultScheduler] fetchTaskGroupsFromDate: yesterday
+                                                          toDate: today
+                                          forTasksMatchingFilter: filterForallTasks
+                                                      usingQueue: [NSOperationQueue mainQueue]
+                                                 toReportResults: ^(NSDictionary *taskGroups, NSError * __unused queryError)
+         {
+             NSArray *actualTaskGroups = taskGroups.allValues.firstObject;
+             if (actualTaskGroups.count > 0) {
+                 
+                 BOOL includeToday = YES;
+                 for (APCTaskGroup *taskGroup in actualTaskGroups) {
+                     if ([taskGroup.dateFullyCompleted.dayBefore isEqualToDate:[NSDate date].dayBefore]) {
+                         // same day
+                         includeToday = NO;
+                         break;
+                     }
+                 }
+                 
+                 [self scheduleDailyRandomReminderIncludeToday:includeToday];
+             }
+         }];  // first fetch:  required tasks, for a range of dates
+    }
+}
+
+- (void)scheduleDailyRandomReminderIncludeToday:(BOOL)includeToday {
+    
+    // Schedule the Task notification
+    UILocalNotification* taskNotification = [[UILocalNotification alloc] init];
+    taskNotification.alertBody = gTaskReminderMessage;
+    
+    NSDate* normalFireDate = [self randomDailyFireDate];
+    
+    // if the fire date is within an hour of current time, then skip for today
+    NSDate *hourFromNow = [[NSDate date] dateByAddingTimeInterval:[self oneMoreHour]];
+    if ([normalFireDate isEarlierThanDate:hourFromNow]) {
+        includeToday = NO;
+    }
+    
+    normalFireDate = includeToday ? normalFireDate : [normalFireDate dateByAddingTimeInterval:[self oneMoreDay]];
+    
+    taskNotification.fireDate = normalFireDate;
+    taskNotification.repeatInterval = NSCalendarUnitDay;
```

Comment:
We could, but i'm not sure it's worth it as the reminders are rescheduled with a new random time each time the app runs (actually, each time the Activities view is shown). If we don't use the repeat interval function and create them manually, how many days out would we schedule? 7, 30, more?